### PR TITLE
86 2d bifurcation diagrams

### PR DESCRIPTION
**Pull request by kmclaren-trinity**

### CODE REVIEW

**Review by sfogarty — COMMENTED** [review]
Make an issue to restore taking a dspoint into apply. It should work, we just have to add y to ....

File: R/dsmodel.R [hunk 1]
```diff
@@ -313,32 +313,33 @@ dsmodel <- function(fun, title="", display = TRUE) {
...
+		  #}
 		  if(!(rangeMult==0 || rangeMult==Inf ||is.null(rangeMult)) && is.null(self$range)){
 		    stop("is.stable with rangeMult!=0 requires range() to have been composed with the model.")
 		  }
 		  #moves all the points untill they are either all infinite, fixed, or outside of range*rangeMult
 		  for(i in 1:numTries) {
-		    startPoint <- self$apply(x,y,iters=initIters,accumulate=FALSE,crop=FALSE)
-		    if(!self$has.diverged(startPoint$x,startPoint$y,rangeMult)){
+		    startPoint <- self$apply(x,y,...,iters=initIters,accumulate=FALSE,crop=FALSE)
+		    if(!self$has.diverged(startPoint[[1]],startPoint[[2]],rangeMult)){
```
Comment (sfogarty) [inline R/dsmodel.R:338]: The inconsistent of [[1]] vs $x is worrisome. What if $x and $y are valid, but in opposite order?

Comment (sfogarty) [reply R/dsmodel.R:338]: Perhaps we should define functions getX and getY? it would refactor a lot of this.

**Review by sfogarty — COMMENTED** [review]
When did the man files get added? that seems wrong.

**Comment by kmclaren-trinity** [comment]
I think I accidentally added the man files with a "git add -A".

File: R/dsmodel.R (hunk 1 above)
Comment (sfogarty) [reply R/dsmodel.R:338]: Make an issue for [[1]] vs $x, quoting the above concern.  Get rid of the rd files. Then re-push and I'll accept the pull request.

**Review by sfogarty — CHANGES_REQUESTED** [review]
Make an issue for [[1]] vs $x, quoting the above concern. Get rid of the rd files. Then re-push and I'll accept the pull request.

**Comment by kmclaren-trinity** [comment]
Changes implemented?

**Review by sfogarty — CHANGES_REQUESTED** [review]
Testing a request changes.

**Review by sfogarty — APPROVED** [review]

**Review by sfogarty — APPROVED** [review]

**Pull request merged**